library/std/src/sys_common/rt.rs

```diff
@@ -0,0 +1,62 @@
+use crate::sync::Once;
+use crate::sys;
+use crate::sys_common::thread_info;
+use crate::thread::Thread;
+
+// One-time runtime initialization.
+// Runs before `main`.
+#[cfg_attr(test, allow(dead_code))]
+pub fn init(argc: isize, argv: *const *const u8) {
+    static INIT: Once = Once::new();
+    INIT.call_once(|| unsafe {
```

Comment:
What's the use of the `Once` static here? Don't we already know for sure this is called exactly once?

Reply:
`sys_common::rt::cleanup` needs to be safe to call multiple times: it could be called in `process::exit`, then panic, which is caught by `lang_start_internal`, and then be called again. For symmetry I then made `sys_common::rt::init` also not unsafe to call, which is why I added the `Once`, to not have to rely on that it can only be called once remaining true in the future. I could remove it however.

Reply:
Yes, `exit()` could also be called from multiple threads concurrently, so `cleanup` definitely needs this. But for `init` I'd remove it if it's not necessary, to not confuse the reader about what is going on.

Reply:
Sure 👍🏻